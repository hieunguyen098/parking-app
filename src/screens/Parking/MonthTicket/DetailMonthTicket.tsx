import { StyleSheet, Text, View, Image, Modal, TouchableOpacity, Button, TextInput } from 'react-native';
import React, { useState } from 'react';
import BottomButton from '../../../components/Buttons/BottomButton';
import FieldValue from '../../../components/FieldValue';
import { GlobalStyles } from '../../../constants';
import SmallButton from '../../../components/Buttons/SmallButton';
import ModalExtendMonthTicket from './ModalExtendMonthTicket';
import ModalGiveTicket from './ModalGiveTicket';
import moment from "moment";

const DetailMonthTicket = ({route} : any) => {
    const { item } = route.params;
    const [extendNumber, setExtendNumber] = useState(0);
    const [expandTicket, setExpandTicket] = useState(false);
    const [giveTicket, setGiveTicket] = useState(false);

    const handleGiveTicket = () => {
        setGiveTicket(true);
    };

    const handleExpandTicket = () => {
        setExpandTicket(true);
        console.log('Expand ticket');
    };

    const convertDueDate = (numOfMonth: string, startDate: string) => {
        if (!isNaN(parseInt(numOfMonth))) {
            return moment(startDate).add(parseInt(numOfMonth), "month").format("DD/MM/YYYY");
        } else {
            return "NaN"
        }
    }

    return (
        <>
            <ModalExtendMonthTicket item = {item} expandTicket={expandTicket} setExpandTicket={setExpandTicket} extendNumber={extendNumber} setExtendNumber={setExtendNumber}/>
            <ModalGiveTicket item={item} giveTicket={giveTicket} setGiveTicket={setGiveTicket} />
            <View style={styles.container}>
                <View style={styles.boxContent}>
                    <View style={styles.top}>
                        <View style={styles.imageContainer}>
                            <Image
                                source={{
                                    uri: item.locationImageUrl ? item.locationImageUrl : "",
                                }}
                                style={styles.image}
                            />
                        </View>
                        <View style={styles.content}>
                            <Text style={styles.title}>{item.locationName}</Text>
                            <SmallButton
                                title={'Tặng'}
                                style={{ width: 80, marginLeft: 'auto' }}
                                onPress={handleGiveTicket}
                            />
                        </View>
                    </View>
                    <FieldValue fieldName="Số tháng đã mua" value={parseInt(item.number) + extendNumber} />
                    <FieldValue fieldName="Địa chỉ" value={item.locationAddress} />
                    <FieldValue fieldName="Ngày mua" value={moment(item.startDate).format("DD/MM/YYYY")} />
                    <FieldValue fieldName="Có giá trị đến hết ngày" value={convertDueDate(item.number, item.startDate)} />
                </View>
            </View>
            <BottomButton title="Gia hạn" onPress={handleExpandTicket} />
        </>
    );
};

export default DetailMonthTicket;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 20,
    },
    boxContent: {
        backgroundColor: '#fff',
        borderRadius: 20,
        padding: 20,
    },
    top: {
        flexDirection: 'row',
        borderBottomWidth: 1,
        borderColor: GlobalStyles.colors.lightGrey100,
        paddingBottom: 10,
    },
    imageContainer: {
        marginRight: 8,
    },
    image: {
        width: 70,
        height: 70,
        borderRadius: 8,
    },
    content: {
        flex: 1,
        justifyContent: 'space-between',
    },
    title: {
        fontSize: 16,
        fontWeight: '500',
    },
});
