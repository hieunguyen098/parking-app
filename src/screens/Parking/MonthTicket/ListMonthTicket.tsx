import { FlatList, StyleSheet, Text, View } from 'react-native';
import React, {useEffect} from 'react';
import MonthTicketItem from '../../../components/MonthTicketItem';
import {useQuery, useQueryClient} from "react-query";
import {getLocations} from "../../../services/location.api";
import {useFocusEffect} from "@react-navigation/native";
import {getMonthCards} from "../../../services/month-card.api";
import {useSelector} from "react-redux";

const ListMonthTicket = () => {
    const user = useSelector((state: any) => state.auth.user);

    const queryClient = useQueryClient();

    const {
        data,
    } = useQuery({
        queryKey: ['month-cards'],
        queryFn: () => {
            return getMonthCards(user.phone);
        },
        select: (data) => data.data? data.data : [],
    });

    useFocusEffect(
        React.useCallback(() => {
            queryClient.fetchQuery('month-cards').then();
        }, []),
    );

    return (
        <FlatList
            data={data}
            renderItem={({ item }) => {
                return <MonthTicketItem item={item} />;
            }}
            keyExtractor={(item) => item.monthCardId}
            style={styles.contentContainer}
        />
    );
};

export default ListMonthTicket;

const styles = StyleSheet.create({
    contentContainer: {
        flex: 1,
    },
});
