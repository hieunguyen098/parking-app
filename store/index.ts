import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './slices/authSlice';
import { generalReducer } from './slices/generalSlice';

export const store = configureStore({
    reducer: {
        auth: authReducer,
        general: generalReducer,
    },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
