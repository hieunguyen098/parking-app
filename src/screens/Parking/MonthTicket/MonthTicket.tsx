import { StyleSheet, Text, View } from 'react-native';
import React from 'react';
import SearchInput from '../../../components/SearchInput/SearchInput';
import ListMonthTicket from './ListMonthTicket';
import styles from '../styles';

const MonthTicket = () => {
    return (
        <View style={styles.container}>
            <SearchInput style={styles.searchInput} placeholder="Tìm kiếm vé tháng" />
            <Text style={styles.title}>Vé tháng của bạn</Text>
            <ListMonthTicket />
        </View>
    );
};

export default MonthTicket;
