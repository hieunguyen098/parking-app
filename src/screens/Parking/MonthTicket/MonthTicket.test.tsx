import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { store } from '../../../../store';
import MonthTicket from './MonthTicket';

jest.useFakeTimers();
test('test month ticket', () => {
    const queryClient = new QueryClient();
    const component = (
        <QueryClientProvider client={queryClient}>
            <Provider store={store}>
                <NavigationContainer>
                    <MonthTicket />
                </NavigationContainer>
            </Provider>
        </QueryClientProvider>
    );

    const tree = render(component);
    expect(tree).toMatchSnapshot();
});
