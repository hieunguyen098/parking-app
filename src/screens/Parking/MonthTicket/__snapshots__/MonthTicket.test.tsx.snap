// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test month ticket 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 2,
        "flexDirection": "row",
        "height": 44,
        "marginVertical": 10,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text />
    <TextInput
      placeholder="Tìm kiếm vé tháng"
      placeholderTextColor="#FF953A"
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 20,
            "marginLeft": 10,
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "700",
        "marginBottom": 10,
      }
    }
  >
    Vé tháng của bạn
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "0",
          "name": "Nhà xe Trường Đại học Bách Khoa TP.HCM",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "1",
          "name": "Nhà xe Hoài Bão 2",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "2",
          "name": "Nhà xe Hoài Bão",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "3",
          "name": "Nhà xe Hoài Bão 2",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "4",
          "name": "Nhà xe Hoài Bão",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "5",
          "name": "Nhà xe Hoài Bão 2",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "6",
          "name": "Nhà xe Hoài Bão",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "7",
          "name": "Nhà xe Hoài Bão 2",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "8",
          "name": "Nhà xe Hoài Bão",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "9",
          "name": "Nhà xe Hoài Bão 2",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "10",
          "name": "Nhà xe Hoài Bão",
        },
        Object {
          "duedate": "31/12/2023",
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
          "key": "11",
          "name": "Nhà xe Hoài Bão 2",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8E8E8",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
                  }
                }
                style={
                  Object {
                    "borderRadius": 8,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginBottom": 8,
                  }
                }
              >
                Nhà xe Trường Đại học Bách Khoa TP.HCM
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#B3B3B3",
                      "fontStyle": "italic",
                      "marginRight": 8,
                      "textAlign": "right",
                    },
                    Object {
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Có giá trị đến hết ngày 
                31/12/2023
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8E8E8",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
                  }
                }
                style={
                  Object {
                    "borderRadius": 8,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginBottom": 8,
                  }
                }
              >
                Nhà xe Hoài Bão 2
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#B3B3B3",
                      "fontStyle": "italic",
                      "marginRight": 8,
                      "textAlign": "right",
                    },
                    Object {
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Có giá trị đến hết ngày 
                31/12/2023
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8E8E8",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
                  }
                }
                style={
                  Object {
                    "borderRadius": 8,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginBottom": 8,
                  }
                }
              >
                Nhà xe Hoài Bão
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#B3B3B3",
                      "fontStyle": "italic",
                      "marginRight": 8,
                      "textAlign": "right",
                    },
                    Object {
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Có giá trị đến hết ngày 
                31/12/2023
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8E8E8",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
                  }
                }
                style={
                  Object {
                    "borderRadius": 8,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginBottom": 8,
                  }
                }
              >
                Nhà xe Hoài Bão 2
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#B3B3B3",
                      "fontStyle": "italic",
                      "marginRight": 8,
                      "textAlign": "right",
                    },
                    Object {
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Có giá trị đến hết ngày 
                31/12/2023
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8E8E8",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
                  }
                }
                style={
                  Object {
                    "borderRadius": 8,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginBottom": 8,
                  }
                }
              >
                Nhà xe Hoài Bão
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#B3B3B3",
                      "fontStyle": "italic",
                      "marginRight": 8,
                      "textAlign": "right",
                    },
                    Object {
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Có giá trị đến hết ngày 
                31/12/2023
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8E8E8",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
                  }
                }
                style={
                  Object {
                    "borderRadius": 8,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginBottom": 8,
                  }
                }
              >
                Nhà xe Hoài Bão 2
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#B3B3B3",
                      "fontStyle": "italic",
                      "marginRight": 8,
                      "textAlign": "right",
                    },
                    Object {
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Có giá trị đến hết ngày 
                31/12/2023
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8E8E8",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
                  }
                }
                style={
                  Object {
                    "borderRadius": 8,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginBottom": 8,
                  }
                }
              >
                Nhà xe Hoài Bão
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#B3B3B3",
                      "fontStyle": "italic",
                      "marginRight": 8,
                      "textAlign": "right",
                    },
                    Object {
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Có giá trị đến hết ngày 
                31/12/2023
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8E8E8",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
                  }
                }
                style={
                  Object {
                    "borderRadius": 8,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginBottom": 8,
                  }
                }
              >
                Nhà xe Hoài Bão 2
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#B3B3B3",
                      "fontStyle": "italic",
                      "marginRight": 8,
                      "textAlign": "right",
                    },
                    Object {
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Có giá trị đến hết ngày 
                31/12/2023
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8E8E8",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
                  }
                }
                style={
                  Object {
                    "borderRadius": 8,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginBottom": 8,
                  }
                }
              >
                Nhà xe Hoài Bão
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#B3B3B3",
                      "fontStyle": "italic",
                      "marginRight": 8,
                      "textAlign": "right",
                    },
                    Object {
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Có giá trị đến hết ngày 
                31/12/2023
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8E8E8",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
                  }
                }
                style={
                  Object {
                    "borderRadius": 8,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginBottom": 8,
                  }
                }
              >
                Nhà xe Hoài Bão 2
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#B3B3B3",
                      "fontStyle": "italic",
                      "marginRight": 8,
                      "textAlign": "right",
                    },
                    Object {
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Có giá trị đến hết ngày 
                31/12/2023
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
